### PR TITLE
fixes #4369 feat(nimbus): add risk mitigation link field to overview form

### LINKED ISSUE #4369: Add risk mitigation checklist textbox + link to Overview form
1) Add the associated text* and input on the “edit overview” page.
2) Display the value in the input if one is already saved for that experiment.
3) Add the field to the mutation query and ensure it saves on “Save”.
4) This is a required field before the user can request a review. Ensure clicking on “missing details in: Overview” will show this field as required.

*Risk mitigation checklist link: https://docs.google.com/document/d/1zfG2g6pYe9aB7ItViQaw8OcOsXXdRUP70zpZoNC2xcA/edit

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-803)
┆Issue Number: EXP-803

### PR DESCRIPTION
**Pull request by jodyheavener**
This PR adds the UI and updated the GQL mutation to set/update an experiment's "Risk Mitigation Checklist Link".

### CODE REVIEW

**Comment by jodyheavener** [comment]
@lmorchard save for regenerating the TS types and one small test update, this worked perfectly 😋

**Comment by jodyheavener** [comment]
Ahh, @lmorchard it looks like the mutation for this is failing with `This field may not be blank`. I think this needs to remain optional until review time. Want me to file a ticket? Or, if it's a simple change I can throw it in this PR.

**Comment by lmorchard** [comment]
> Ahh, @lmorchard it looks like the mutation for this is failing with This field may not be blank. I think this needs to remain optional until review time. Want me to file a ticket? Or, if it's a simple change I can throw it in this PR.
>
> Edit: can I just add , allow_blank=True to risk_mitigation_link in the v5 serializer? That seems to work

Oof, yeah - I thought `required=False` would do the trick. But if `allow_blank=True` resolves the issue, go ahead and add it to this PR and I can follow up with a test asserting that it passes validation when blank

**Comment by lmorchard** [comment]
Or, if you're feeling frisky, here's what I think the change & test would look like:

```diff
diff --git a/app/experimenter/experiments/api/v5/serializers.py b/app/experimenter/experiments/api/v5/serializers.py
index 30736ba5..70e30ece 100644
--- a/app/experimenter/experiments/api/v5/serializers.py
+++ b/app/experimenter/experiments/api/v5/serializers.py
@@ -318,7 +318,7 @@ class NimbusExperimentUpdateSerializer(
     )
     hypothesis = serializers.CharField(min_length=0, max_length=1024, required=False)
     risk_mitigation_link = serializers.URLField(
-        min_length=0, max_length=255, required=False
+        min_length=0, max_length=255, required=False, allow_blank=True
     )
     documentation_links = NimbusDocumentationLinkSerializer(many=True, required=False)
     reference_branch = NimbusBranchSerializer(required=False)
diff --git a/app/experimenter/experiments/tests/api/v5/test_serializers.py b/app/experimenter/experiments/tests/api/v5/test_serializers.py
index b56418b6..5a846c33 100644
--- a/app/experimenter/experiments/tests/api/v5/test_serializers.py
+++ b/app/experimenter/experiments/tests/api/v5/test_serializers.py
@@ -46,7 +46,6 @@ class TestCreateNimbusExperimentOverviewSerializer(TestCase):
             "name": "Test 1234",
             "hypothesis": "Test hypothesis",
             "application": NimbusExperiment.Application.DESKTOP.value,
-            "risk_mitigation_link": "https://example.com/risk",
             "public_description": "Test description",
         }
 
@@ -60,11 +59,23 @@ class TestCreateNimbusExperimentOverviewSerializer(TestCase):
         self.assertEqual(experiment.name, data["name"])
         self.assertEqual(experiment.application, data["application"])
         self.assertEqual(experiment.hypothesis, data["hypothesis"])
-        self.assertEqual(experiment.risk_mitigation_link, data["risk_mitigation_link"])
         self.assertEqual(experiment.public_description, data["public_description"])
         # Owner should match the email of the user who created the experiment
         self.assertEqual(experiment.owner, self.user)
 
+    def test_serializer_accepts_blank_risk_mitigation_link(self):
+        data = {
+            "name": "Test 1234",
+            "hypothesis": "Test hypothesis",
+            "application": NimbusExperiment.Application.DESKTOP.value,
+            "public_description": "Test description",
+            "risk_mitigation_link": "",
+        }
+        serializer = NimbusExperimentUpdateSerializer(
+            data=data, context={"user": self.user}
+        )
+        self.assertTrue(serializer.is_valid())
+
     def test_serializer_rejects_bad_name(self):
         data = {
             "name": "&^%&^%&^%&^%^&%^&",
```

**Review by lmorchard — APPROVED** [review]
Pretty straightforward, I'd say we could merge this and correct the `allow_blank` thing in another PR actually

**Comment by lmorchard** [comment]
Since this went quicker than I thought and I don't think it should block this PR as such, hopefully this followup does the trick:

https://github.com/mozilla/experimenter/pull/4416

File: app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx [hunk 1]
```diff
@@ -22,11 +23,15 @@ type FormOverviewProps = {
   onNext?: (ev: React.FormEvent) => void;
 };
 
+export const RISK_MITIGATION_TEMPLATE_LINK =
+  "https://docs.google.com/document/d/1zfG2g6pYe9aB7ItViQaw8OcOsXXdRUP70zpZoNC2xcA/edit";
```
[the remark this answers is not in the document]
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx:27]: Agree. https://jira.mozilla.com/browse/EXP-826

**Pull request merged**